install.ps1

```diff
@@ -39,16 +39,54 @@ $repo = "onflow/flow-cli"
 $versionURL = "https://api.github.com/repos/$repo/releases/latest"
 $assetsURL = "https://github.com/$repo/releases/download"
 
-# Add the GitHub token to the web request headers if it was provided
-$webRequestOptions = if ($githubToken) {
-	@{ 'Headers' = @{ 'Authorization' = "Bearer $githubToken" } }
-} else {
-    @{}
+# Function to get the latest version
+function Get-Version {
+    param (
+        [string]$repo,
+        [string]$searchTerm,
+        [string]$githubTokenHeader
+    )
+
+    $page = 1
+    $version = null
+
+    while (-not $version) {
+        if ($githubTokenHeader) {
+            $response = Invoke-WebRequest -Uri "https://api.github.com/repos/$repo/releases?per_page=100&page=$page" -Headers @{ 'Authorization' = $githubTokenHeader } -UseBasicParsing -ErrorAction SilentlyContinue
+        } else {
+            $response = Invoke-WebRequest -Uri "https://api.github.com/repos/$repo/releases?per_page=100&page=$page" -UseBasicParsing -ErrorAction SilentlyContinue
+        }
+
+        $status = $response.StatusCode
+
+        if ($status -eq 403 -and $githubTokenHeader) {
+            Write-Output "Failed to get latest version from Github API, is your GITHUB_TOKEN valid? Trying without authentication..."
+            $githubTokenHeader = ""
+            continue
+        }
+
+        if ($status -ne 200) {
+            Write-Output "Failed to get latest version from Github API, please manually specify a version to install as an argument to this script."
+            return $null
+        }
+
+        $jsonResponse = $response.Content | ConvertFrom-Json
+
+        foreach ($release in $jsonResponse) {
+            if ($release.tag_name -like "*$searchTerm*") {
+                $version = $release.tag_name
+                break
```

Comment:
What if no tag is found?

Reply:
If no tag is found, it will reach the end of the outer loop, call `$page++`, and iterate again.  Once it runs out of pages (`$status -ne 200`), it will return null.